src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Helper.java

```diff
@@ -347,4 +349,52 @@ public static void populateDynamicListOffices(CaseData caseData, String caseType
                 : new ArrayList<>();
     }
 
+    public static CaseData populateDynamicRespondentRepresentativeNames(CaseData caseData) {
+        List<DynamicValueType> listItems = createDynamicRespondentName(caseData.getRespondentCollection());
+        if (!listItems.isEmpty()) {
+            var dynamicFixedListType = new DynamicFixedListType();
+            dynamicFixedListType.setListItems(listItems);
+            if (caseData.getRepCollection() != null && !caseData.getRepCollection().isEmpty()) {
+                ListIterator<RepresentedTypeRItem> repItr = caseData.getRepCollection().listIterator();
+                int index;
+                while (repItr.hasNext()) {
+                    index = repItr.nextIndex() + 1;
```

Comment:
If you need to use indexes of lists then use a for loop rather than an iterator. Using both an iterator and an index variable makes your code harder to read

Reply:
Removed index